x-pack/plugins/canvas/server/usage/workpad_collector.ts

```diff
@@ -21,25 +74,30 @@ export function handleResponse({ hits }) {
   const workpadsInfo = workpadDocs.map(hit => {
     const workpad = hit._source[CANVAS_TYPE];
 
-    let pages;
+    let pages = { count: 0 };
     try {
       pages = { count: workpad.pages.length };
     } catch (err) {
+      // eslint-disable-next-line
       console.warn(err, workpad);
     }
     const elementCounts = workpad.pages.reduce(
       (accum, page) => accum.concat(page.elements.length),
-      []
+      [] as number[]
     );
-    const functionCounts = workpad.pages.reduce((accum, page) => {
-      return page.elements.map(element => {
-        const ast = fromExpression(element.expression);
-        collectFns(ast, cFunction => {
-          functionSet.add(cFunction);
+    const functionCounts = workpad.pages.reduce(
+      (accum, page) => {
+        return page.elements.map(element => {
+          const ast = fromExpression(element.expression) as AST;
+          collectFns(ast, cFunction => {
+            functionSet.add(cFunction);
+          });
+          return ast.chain.length; // get the number of parts in the expression
         });
-        return ast.chain.length; // get the number of parts in the expression
-      });
-    }, []);
+      },
+      [] as number[]
```

Comment:
*Always* avoid `as`... it's a sure-fire sign something isn't right.  In this case, `reduce` can be supplied a generic, which is defaulted to `never[]`:

```
/**
      * Calls the specified callback function for all the elements in an array. The return value of the callback function is the accumulated result, and is provided as an argument in the next call to the callback function.
      * @param callbackfn A function that accepts up to four arguments. The reduce method calls the callbackfn function one time for each element in the array.
      * @param initialValue If initialValue is specified, it is used as the initial value to start the accumulation. The first call to the callbackfn function provides this value as an argument instead of an array value.
      */
    reduce<U>(callbackfn: (previousValue: U, currentValue: T, currentIndex: number, array: T[]) => U, initialValue: U): U;
```

So, all you need to do is supply the generic yourself:

```
const functionCounts = workpad.pages.reduce<number[]>((accum, page) => {
```

Same applies to line 86.